onnxruntime/contrib_ops/cuda/bert/ngram_repeat_block_impl.cu

```diff
@@ -20,10 +20,15 @@ using namespace onnxruntime::cuda;
 __global__ void banRepeatedTokens(const int64_t* __restrict__ tokens,
                                   float* __restrict__ lprobs,
                                   int max_predict_len, int vocab_size,
+                                  int banned_no_further_than,
                                   int no_repeat_ngram_size) {
   auto row = blockIdx.x;
   auto col = threadIdx.x;
   auto start = row * (max_predict_len) + col;
+  auto cur_len = (row + 1) * (max_predict_len);
+  if (banned_no_further_than > 0 && cur_len > banned_no_further_than_){
+    start = cur_len - banned_no_further_than_;
```

Comment:
You shouldn't change `start`. cuda kernel runs in parallel, so what you can do is to skip the action if the `start` of this thread is not within the range.
```c++
bool skip_this_ngram = ...;
if (!skip_this_ngram) {
  // old code before sync
}
__syncthreads();
if (!skip_this_ngram) {
  // old code after sync
}
```

Reply:
Done